contents/security/sqlInjection.md

```diff
@@ -0,0 +1,149 @@
+# SQL Injection #
+
+Author: [Lewis Koh](https://github.com/nus-cs3281/2018/blob/master/students/lewisKoh/lewisKoh-Resume.md)
+
+## Overview ##
+
+SQL (Structured Query Language) Injection is a vulnerability found in websites using databases.
+
+SQL is a common language which is used by websites to communicate with databases. Databases 
+can be used to store persistent data, such as usernames and passwords, sensitive account data, 
+or other important information used by the website. Databases are usually made of many "tables" 
+which are organised in rows and columns. Each row is a separate entry in the table, and 
+each column is a specific parameter which can be used by the entry.
+
+SQL is used to interact with the database by sending "queries" which the database responds to. 
+Some common SQL commands used include: 
+* SELECT - retrieves information from a table
+* UPDATE - changes information from a table
+* INSERT INTO - adds a new entry into a table
+* DELETE FROM - deletes information from a table
+
+Websites combine SQL syntax with user input to query the underlying database of the website. 
+For example, blog websites may ask for a username and password from users for them to log 
+in to the site. This input is used to create the SQL statement that will be sent to the database. 
+A website which is vulnerable to SQL injection attacks may construct the SQL query like this:
+
+```sql
+String query = "SELECT * FROM Users WHERE Username = ‘" + input_username +"’ 
+                AND Password = ‘" + input_password + "’";
+```
+
+As an example, if a user entered their username, "Alice", and password, "password" into the website 
+to try to gain access, the constructured query would look like this:
+
+```sql
+SELECT * FROM Users WHERE Username = ‘Alice’ 
+AND Password = ‘password’
+```
+
+This query would look at the "Users" table in the database, and return any entries where the 
+column "Username" matches "Alice" and the column "Password" matches "password". 
+The intention may be to allow access to the website if the query returns any entry, as a result 
+would imply a matching username and password.
+
+
+However, are users limited to just filling in the parameters? Some websites directly use the input given, 
+thus the users are able to modify the query, or even to add in their own. This can easily be used 
```

Comment:
I would say `SQL query` to be more precise. Also, the next sentence can be joined here as `for malicious purposes` to make the paragraph more concise. Because, for a  newbie, the high level description is quite cryptic, so I think it's good to make this part concise and quickly introduce the example to illustrate the point.